samza-core/src/main/java/org/apache/samza/runtime/RemoteApplicationRunner.java

```diff
@@ -57,13 +55,21 @@
    * @param config configuration for the application
    */
   public RemoteApplicationRunner(SamzaApplication app, Config config) {
-    this.appDesc = ApplicationDescriptorUtil.getAppDescriptor(app, config);
-    this.planner = new RemoteJobPlanner(appDesc);
+    this.app = app;
+    this.config = config;
   }
 
   @Override
   public void run(ExternalContext externalContext) {
+    if (new JobConfig(config).getConfigLoaderFactory().isPresent()) {
+      JobRunner runner = new JobRunner(config);
+      runner.submit();
+      return;
+    }
+
+    // TODO: Clean this up once SAMZA-2405 is completed when legacy flow is removed.
```

Comment:
Would it be helpful to create a ticket for this cleanup work (for all changes related to the SEP) so we don't lose track of it?

Reply:
Good point, I will create a ticket for it.